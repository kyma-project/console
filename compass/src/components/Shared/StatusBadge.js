import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from 'fundamental-react/lib/Badge';

class StatusBadge extends React.Component {
  PropTypes = {
    status: PropTypes.string,
  };

  processStatus() {
    const { status } = this.props;
    let type = 'warning';

    switch (status) {
      case 'INITIAL':
        return <Badge>{status}</Badge>;
       case 'READY':
        type = 'success';
        break;
      case 'UNKNOWN':
        type = 'warning';
        break;
      case 'FAILED':
        type = 'error';
        break;
      default:
        type = 'warning';
    }

    return <Badge type={type}>{status}</Badge>;
  }

  render = () => (
    this.processStatus()
  );
}

export default StatusBadge