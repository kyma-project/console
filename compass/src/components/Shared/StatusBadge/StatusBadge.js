import React from 'react';
import { Badge } from 'fundamental-react/Badge';

const StatusBadge = ({ status }) => {
  let type;

  switch (status) {
    case 'INITIAL':
      return <Badge>{status}</Badge>;
    case 'READY':
      type = 'success';
      break;
    case 'UNKNOWN':
      type = 'warning';
      break;
    case 'FAILED':
      type = 'error';
      break;
    default:
      type = 'warning';
  }

  return <Badge type={type}>{status}</Badge>;
};

export default StatusBadge;
