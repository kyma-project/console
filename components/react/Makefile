ci-pr: resolve build validate test
ci-master: resolve build validate test
ci-release: resolve build validate test

resolve:
	npm install

build:
	npm run build

validate:
	npm run lint-check
	eslint -c .eslinterrc ./src

test:
	CI=true npm run test
	