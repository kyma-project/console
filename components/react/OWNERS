reviewers:
    - default-reviewers

approvers:
    - default-approvers
    - Wawrzyn321
    - content-reviewers