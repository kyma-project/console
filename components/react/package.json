{
  "name": "@kyma-project/react-components",
  "version": "0.0.43",
  "main": "dist/index.js",
  "private": false,
  "license": "Apache-2.0",
  "directories": {
    "dist": "src"
  },
  "scripts": {
    "start:kyma": "npm run build:watch",
    "start:kyma:api": "npm run build:watch",
    "build": "webpack --config webpack.config.prod.js",
    "build:watch": "webpack -w --config webpack.config.dev.js",
    "test": "jest src/**; exit 0",
    "test:watch": "jest src/** --watch",
    "test:watchAll": "jest src/** --watchAll",
    "test:coverage:report": "open ./coverage/lcov-report/index.html",
    "lint-fix": "prettier --config ../../.prettierrc --write \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\"",
    "lint-check": "prettier --config ../../.prettierrc --check \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\""
  },
  "dependencies": {
    "@babel/runtime-corejs2": "^7.12.5",
    "lodash": "^4.17.21",
    "polished": "^1.8.1",
    "styled-components-modifiers": "^1.0.1"
  },
  "devDependencies": {
    "@babel/types": "7.0.0-beta.44",
    "ajv": "^6.12.3",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.1",
    "babel-jest": "^22.4.1",
    "babel-loader": "^7.1.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "exec-sh": "^0.3.3",
    "file-loader": "^1.1.11",
    "fiori-fundamentals": "1.5.3",
    "fontfaceobserver": "^2.0.13",
    "handlebars": "^4.7.3",
    "jest": "^24.0.0",
    "jest-styled-components": "^4.11.0-1",
    "npm-run-all": "^4.1.1",
    "raf": "^3.4.0",
    "react-jsonschema-form": "1.7.0",
    "react-modal": "^3.5.1",
    "react-scrollbar": "^0.5.4",
    "react-test-renderer": "^16.0.0",
    "style-loader": "^0.20.2",
    "styled-components": "^3.2.3",
    "styled-components-grid": "^2.2.0",
    "webpack": "^4.28.1",
    "webpack-cli": "^3.2.1",
    "webpack-merge": "^4.2.1"
  },
  "peerDependencies": {
    "styled-components": "^3.2.3"
  },
  "resolutions": {
    "uglify-es": "3.3.7"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "src/**",
      "!src/**/*.snap",
      "!src/components/index.js",
      "!src/modifiers/**",
      "!src/index.js"
    ],
    "moduleNameMapper": {
      "^.+\\.(css|less|scss)$": "babel-jest"
    },
    "moduleDirectories": [
      "node_modules",
      "./"
    ],
    "setupFiles": [
      "raf/polyfill",
      "./test_setup/config"
    ]
  }
}
