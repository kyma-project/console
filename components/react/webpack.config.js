const webpack = require('webpack');
const path = require('path');
const pkg = require('./package.json');

require('babel-polyfill');

let libraryName = pkg.name;

module.exports = {
  entry: ['./src/index'],
  module: {
    loaders: [
      { test: /\.js?$/, loader: 'babel-loader', exclude: /node_modules/ },
      {
        test: /\.css$/,
        use: [{ loader: 'style-loader' }, { loader: 'css-loader' }],
      },
      {
        test: /\.(eot|svg|ttf|woff|woff2|otf)$/,
        loader: 'file-loader?name=fonts/[name].[ext]',
      },
    ],
  },
  resolve: {
    modules: [path.resolve('./node_modules'), 'node_modules'],
    extensions: ['.js'],
  },
  output: {
    path: path.join(__dirname, '/lib'),
    publicPath: '/',
    filename: 'index.js',
    library: libraryName,
    libraryTarget: 'umd',
    umdNamedDefine: true,
  },
  devtool: 'source-map',
  devServer: {
    contentBase: './src',
    hot: true,
  },
  plugins: [
    new webpack.optimize.OccurrenceOrderPlugin(),
    new webpack.HotModuleReplacementPlugin(),
    new webpack.NoEmitOnErrorsPlugin(),
  ],
  externals: {
    'styled-components': {
      commonjs: 'styled-components',
      commonjs2: 'styled-components',
      amd: 'styled-components',
    },
  },
};
