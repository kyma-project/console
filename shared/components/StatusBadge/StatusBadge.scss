.fd-badge.status-badge {
  font-size: var(--sapFontSize, 0.875rem);
  font-family: var(
    --sapFontFamily,
    '72',
    '72full',
    Arial,
    Helvetica,
    sans-serif
  );
  margin: 0;

  max-width: 100%;
  display: inline-block;

  line-height: 1;

  //inverted
  height: auto;
  min-height: 1rem;
  width: auto;
  padding: 0.14rem 0.4rem;
  border-radius: 0.125rem;
  font-weight: 700;
  background-color: var(--sapButton_Neutral_Background, #6a6d70);
  color: var(--sapButton_Neutral_TextColor, #fff);

  &--success {
    background-color: var(--sapButton_Success_Background, #107e3e);
  }
  &--warning {
    background-color: var(--sapButton_Critical_Background, #e9730c);
  }
  &--error {
    background-color: var(--sapButton_Negative_Background, #b00);
  }
  &--info {
    background-color: var(--sapInformativeElementColor, #0a6ed1);
  }

  &.hasTooltip {
    cursor: help;
  }
}
