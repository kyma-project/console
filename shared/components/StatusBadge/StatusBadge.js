import React from 'react';
import { Badge } from 'fundamental-react/Badge';
import PropTypes from 'prop-types';
import './StatusBadge.scss';
import classNames from 'classnames';
import { Tooltip } from '../Tooltip/Tooltip';

const resolveType = status => {
  switch (status.toUpperCase()) {
    case 'INITIAL':
      return 'info';

    case 'READY':
    case 'RUNNING':
    case 'SUCCESS':
      return 'success';

    case 'UNKNOWN':
    case 'WARNING':
      return 'warning';

    case 'FAILED':
    case 'ERROR':
    case 'FAILURE':
      return 'error';

    default:
      return undefined;
  }
};

export const StatusBadge = ({
  status,
  type,
  children,
  autoResolveType = false,
  tooltipProps = {},
}) => {
  if (autoResolveType) type = resolveType(status);

  const classes = classNames('status-badge', {
    ['status-badge--' + type]: type,
    'has-tooltip': children,
  });

  const badgeElement = (
    <Badge role="status" modifier="filled" className={classes}>
      {status}
    </Badge>
  );

  if (children)
    return (
      <Tooltip content={children} {...tooltipProps}>
        {badgeElement}
      </Tooltip>
    );
  else return badgeElement;
};

StatusBadge.propTypes = {
  status: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  autoResolveType: PropTypes.bool,
};
