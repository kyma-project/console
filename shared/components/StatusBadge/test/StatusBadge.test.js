import React from 'react';
import { StatusBadge } from '../StatusBadge';
import { render } from '@testing-library/react';

describe('StatusBadge', () => {
  it('renders status text', () => {
    const { queryByText } = render(<StatusBadge status="INITIAL" />);

    expect(queryByText('INITIAL')).toBeInTheDocument();
  });

  it('displays help cursor when tooltip content is set', () => {
    const { getByRole } = render(
      <StatusBadge status="INITIAL">some content</StatusBadge>,
    );
    expect(getByRole('status')).toHaveStyle(`cursor: help`);
  });
});
