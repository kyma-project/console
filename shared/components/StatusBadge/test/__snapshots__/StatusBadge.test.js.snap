// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadge displays warning when autoResolveType is set and "children" is a node 1`] = `
Array [
  "'autoResolveType' prop requires a non-string status value",
]
`;
